SlowCheetah.JDT/JsonTransformation.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Sayed Ibrahim Hashimi. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See  License.md file in the project root for full license information.
+
+namespace SlowCheetah.JDT
+{
+    using System;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Represents a transformation based on a JSON file using JDT
+    /// </summary>
+    public class JsonTransformation
```

Comment:
I am not really sold on the usefulness of `JsonDocument` and `JsonTransformation`. All they are is adding the file on disk information with the `JObject`. I do not think JDT should be responsible for reading/writing JSON from/to disk - leave that to the consumer of this API to decide how to do that. We can either work directly with `JObject` or if we want to not have newtonsoft in our public API then we can operate on streams.

As for eventually needing the file name during logging, this can be solved another way. Perhaps an overload for `JdtProcessor.ProcessTransform` with a _logging context_ style object that contains the names of the source and transform files.